lib/oli_web/live/community_live/associated/new_view.ex

```diff
@@ -0,0 +1,129 @@
+defmodule OliWeb.CommunityLive.Associated.NewView do
+  use Surface.LiveView, layout: {OliWeb.LayoutView, "live.html"}
+  use OliWeb.Common.SortableTable.TableHandlers
+
+  alias Oli.Authoring.Course
+  alias Oli.Groups
+  alias OliWeb.Common.{Breadcrumb, Filter, Listing}
+  alias OliWeb.CommunityLive.Associated.{IndexView, TableModel}
+  alias Oli.Delivery.Sections.Blueprint
+  alias OliWeb.Router.Helpers, as: Routes
+
+  data title, :string, default: "Community Associate New"
+  data breadcrumbs, :any
+
+  data query, :string, default: ""
+  data total_count, :integer, default: 0
+  data offset, :integer, default: 0
+  data limit, :integer, default: 20
+  data sort, :string, default: "sort"
+  data page_change, :string, default: "page_change"
+
+  @table_filter_fn &__MODULE__.filter_rows/3
+  @table_push_patch_path &__MODULE__.live_path/2
+
+  def filter_rows(socket, query, _filter) do
+    Enum.filter(socket.assigns.sources, fn a ->
+      String.contains?(String.downcase(TableModel.get_field(:title, a)), String.downcase(query))
+    end)
+  end
+
+  def live_path(socket, params) do
+    Routes.live_path(socket, __MODULE__, socket.assigns.community_id, params)
+  end
+
+  def breadcrumb(community_id) do
+    IndexView.breadcrumb(community_id) ++
+      [
+        Breadcrumb.new(%{
+          full_title: "New",
+          link: Routes.live_path(OliWeb.Endpoint, __MODULE__, community_id)
+        })
+      ]
+  end
+
+  defp retrieve_all_sources() do
+    (Course.list_projects() ++ Blueprint.list())
+    |> Enum.with_index(fn element, index ->
+      type =
+        case Map.has_key?(element, :type) and Map.get(element, :type) == :blueprint do
+          true -> "product"
+          _ -> "project"
+        end
+
+      Map.merge(element, %{
+        unique_type: type,
+        unique_id: index
+      })
+    end)
+  end
+
+  def mount(%{"community_id" => community_id}, _session, socket) do
+    sources = retrieve_all_sources()
```

Comment:
We are not filtering the projects/products that the community already has, so if you try to add an existing one, an error message will be displayed stating that. Is that correct or do we want to filter or mark them in some way? If so, we must think carefully about how, because having thousands could become an expensive operation

Reply:
Lets filter now - there are many way to handle that so it doesn't become expensive.